### DISCUSSION TITLE
使用1.8版本使用spark.master=yarn，spark.submit.deployMode=cluster启动引擎建表报错

### DISCUSSION THREAD

**Discussion by sunnyboy3** [apache/kyuubi#6111] [Q&A]
使用如下命令启动引擎
`bin/beeline  -u 'jdbc:hive2://xx.xx.xx:10009/default;#kyuubi.session.real.user=wp;kyuubi.engine.jdbc.driver.class=com.mongodb.spark.sql;spark.driver.extraClassPath=/opt/apache-kyuubi-1.8.0-bin/spark-driver-jars/mongo-spark-connector_2.12-2.4.1.jar:/opt/apache-kyuubi-1.8.0-bin/spark-driver-jars/mongo-java-driver-3.11.0-rc0.jar;spark.jars=/opt/apache-kyuubi-1.8.0-bin/spark-driver-jars/mongo-spark-connector_2.12-2.4.1.jar,/opt/apache-kyuubi-1.8.0-bin/spark-driver-jars/mongo-java-driver-3.11.0-rc0.jar;spark.yarn.jars=hdfs://nameservice1/dolphinscheduler-v2/default/resources/spark-jars/*.jar;spark.submit.deployMode=cluster'`
启动成功
但是执行表的创建
```
CREATE TABLE IF NOT EXISTS spark_catalog.default.student_question_min(
_id STRING, 
classType STRING, 
difficulty STRING,
grade STRING,
knowledgePoint STRING,
questionKind STRING,
questionNum INT,
questionScore DOUBLE,
studentName STRING,
studentNo LONG,
studentScore DOUBLE,
teacherName STRING,
totalScore DOUBLE,
group INT
) USING com.mongodb.spark.sql
OPTIONS (
'spark.mongodb.input.uri'='mongodb://admin:xxxxxx', 
'spark.mongodb.output.uri'='mongodb://admin:xxxxx',
'spark.mongodb.input.batchSize'=100000,
'spark.mongodb.input.partitioner'='MongoPaginateByCountPartitioner',
'spark.mongodb.input.partitionerOptions.partitionKey'='_id',
'spark.mongodb.input.partitionerOptions.numberOfPartitions'=32
);
```
报错如下：
```
24/02/28 14:19:19 ERROR ExecuteStatement: Error operating ExecuteStatement: java.lang.NoClassDefFoundError: org/apache/spark/sql/sources/DataSourceRegister
        at java.lang.ClassLoader.defineClass1(Native Method)
        at java.lang.ClassLoader.defineClass(ClassLoader.java:763)
        at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
        at java.net.URLClassLoader.defineClass(URLClassLoader.java:467)
        at java.net.URLClassLoader.access$100(URLClassLoader.java:73)
```

我怀疑是jar包没有下载，麻烦帮我看看。

**Comment by pan3793** [comment]
there are multiple components/processes in a distribution system, which component/process reports the error? and any more stacktraces?

**Reply by pan3793** [reply]
additionally, I have no idea why you use such a JDBC URL configuration combination, and have no sense of your cluster setup
